src/MiddlewareRegistry.ts

```diff
@@ -40,32 +40,60 @@ export class MiddlewareRegistry<R extends MiddlewareRequest> {
    */
   public async execute() {
     const middlewareChain = this.composeMiddlewareChain()
-    let middlewareExitCode: MiddlewareExitCode = MiddlewareExitCode.NEXT
+    let middlewareExitCode: MiddlewareExitCode = MiddlewareExitCode.NEXT_IN_CHAIN
     let middlewareFunction = middlewareChain.next()
     do {
-      middlewareExitCode = (await middlewareFunction.value(this.request)) || MiddlewareExitCode.NEXT
-      middlewareFunction = middlewareChain.next()
-    } while (middlewareExitCode !== MiddlewareExitCode.EXIT)
+      if (Array.isArray(middlewareFunction?.value)) {
```

Comment:
Can this Elvis operator be removed? I don't see why we'd anticipate the key not existing on `middlewareFunction`

Reply:
You're right, I think this was a remnant of troubleshooting. Removed